### PR TITLE
Create an RGCN model using cugraph-ops

### PR DESCRIPTION
**Pull request by tingyu66**
This PR adds an accelerated R-GCN model that uses the aggregation functions in `pylibcugraphops`, a Python wrapper of `cugraph-ops`.
Fixes https://github.com/rapidsai/cugraph-ops/issues/180.

### Changes
- Add `torch_geometric.nn.RGCNConvCugraphOps` model
- Add an example of its usage in mini-batch training (using `torch_geometric.loader.NeighborLoader`)
- "Unit" test that verifies forward and backward computation from `RGCNConvCugraphOps` against `RGCNConv`

### Performance
Hardware: 32-core AMD Ryzen Threadripper Pro 3975WX, NVIDIA RTX A6000
Benchmark: Mini-batch `rgcn_cugraphops.py` example:
Parameters: batch_size = 128, num_neighbors = [10,10], num_bases = 10, aggr = 'sum', h_dim = 16

The table below reports the total training time for 100 epochs and the test accuracy for each dataset.

| Dataset   | time (s) | accuracy |
| ------------ | --------- | ----------- |
| aifb         |   1.92   |   ~0.92  | 
| mutag     |   2.01   |   ~0.65  |
| bgs         |   1.26   |   ~0.68  |
| am          |  14.14  |   ~0.74  |

### CODE REVIEW

**Comment by tingyu66** [comment]
Note: I haven't found a way in PyG models to let initial input features be trainable torch parameter when using mini-batch training. Therefore, I used cugraph-ops' RGCN in a full-graph fashion for the following performance comparison, so that the computational work from using both models is identical.

Dataset: AIFB, num_relations = 90, num_edges = 58086, num_nodes = 8285
Parameters: num_bases = 20, num_neighbors = 1247

Timings below reflect the average forward and backward time of the model in one epoch, and they are profiled with nvtx.

| model   | forward time (ms) | backward time (ms)|
| ------------ | --------- | ----------- |
| RGCNConv  |   35.2   |   18.5  | 
| FastRGCNConv     |  0.6   |   5.0  |
| CuGraphOpsRGCNConv  |   11.0   |   0.7  |

**Comment by tingyu66** [comment]
Closed as this is superceded by: https://github.com/pyg-team/pytorch_geometric/pull/6278
